nemo/collections/asr/models/transformer_bpe_models.py

```diff
@@ -297,7 +367,9 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
                 config,
                 global_rank=self.global_rank,
                 world_size=self.world_size,
-                dataset=LhotseSpeechToTextBpeDataset(tokenizer=self.tokenizer,),
+                dataset=LhotseSpeechToTextBpeDataset(
```

Comment:
Wondering if we can just replace `LhotseSpeechToTextBpeDataset` with `CanaryDataset` mentioned above or if we want to perform some special detection which class to use; another option is to subclass this transformer class into sth like `canary.py` and override all methods updated here?